crates/jxl-frame/src/data/spline.rs

```diff
@@ -1,26 +1,371 @@
+        }
+
+        Ok(Self {
+            quant_adjust,
+            quant_splines: splines,
+        })
+    }
+}
+
+impl QuantSpline {
+    fn new(start_point: (i32, i32)) -> Self {
+        Self {
+            start_point,
+            points_deltas: Vec::new(),
+            xyb_dct: [[0; 32]; 3],
+            sigma_dct: [0; 32],
+        }
+    }
+
+    fn decode<R: Read>(
+        &mut self,
+        decoder: &mut Decoder,
+        bitstream: &mut Bitstream<R>,
+        num_pixels: usize,
+    ) -> Result<()> {
+        let num_points = decoder.read_varint(bitstream, 3)? as usize;
+
+        let max_num_points = usize::min(MAX_NUM_CONTROL_POINTS, num_pixels / 2);
+        if num_points > max_num_points {
+            return Err(crate::Error::TooManySplinePoints(num_points));
+        }
+
+        self.points_deltas.resize(num_points, (0, 0));
+
+        for delta in &mut self.points_deltas {
+            delta.0 = unpack_signed(decoder.read_varint(bitstream, 4)?);
+            delta.1 = unpack_signed(decoder.read_varint(bitstream, 4)?);
+        }
+        for color_dct in &mut self.xyb_dct {
+            for i in color_dct {
+                *i = unpack_signed(decoder.read_varint(bitstream, 5)?);
+            }
+        }
+        for i in &mut self.sigma_dct {
+            *i = unpack_signed(decoder.read_varint(bitstream, 5)?);
+        }
+        Ok(())
+    }
+
+    pub fn dequant(
+        &self,
+        quant_adjust: i32,
+        base_correlations_xb: Option<(f32, f32)>,
+        estimated_area: &mut u64,
+    ) -> Spline {
+        let mut manhattan_distance = 0;
+        let mut points = Vec::with_capacity(self.points_deltas.len() + 1);
+
+        let mut cur_value = self.start_point;
+        points.push(Point::new(cur_value.0 as f32, cur_value.1 as f32));
+        let mut cur_delta = (0, 0);
+        for delta in &self.points_deltas {
+            cur_delta.0 += delta.0;
+            cur_delta.1 += delta.1;
+            manhattan_distance += cur_delta.0.abs() + cur_delta.1.abs();
+            cur_value.0 += cur_delta.0;
+            cur_value.1 += cur_delta.1;
+            points.push(Point::new(cur_value.0 as f32, cur_value.1 as f32));
+        }
+
+        let mut xyb_dct = [[0f32; 32]; 3];
+        let mut sigma_dct = [0f32; 32];
+        let mut width_estimate = 0f32;
+
+        let quant_adjust = quant_adjust as f32;
+        let inverted_qa = if quant_adjust >= 0.0 {
+            1.0 / (1.0 + quant_adjust / 8.0)
+        } else {
+            1.0 - quant_adjust / 8.0
+        };
+
+        const CHANNEL_WEIGHTS: [f32; 4] = [0.0042, 0.075, 0.07, 0.3333];
+        for chan_idx in 0..3 {
+            for i in 0..32 {
+                xyb_dct[chan_idx][i] =
+                    self.xyb_dct[chan_idx][i] as f32 * CHANNEL_WEIGHTS[chan_idx] * inverted_qa;
+            }
+        }
+        let (corr_x, corr_b) = match base_correlations_xb {
+            Some(val) => (val.0, val.1),
+            None => (0.0, 1.0),
+        };
```

Comment:
This is a common pattern in Rust, and `Option<T>` has a method for this:
```suggestion
        let (corr_x, corr_b) = base_correlations_xb.unwrap_or((0.0, 1.0));
```

I'm quite sure clippy will suggest this if you write `val` in the `Some(val)` match arm.